src/Desktop.Analyzers/UnitTests/DoNotUseInsecureDTDProcessingUseXmlReaderForXPathDocumentTests.cs

```diff
@@ -0,0 +1,366 @@
+            }
+            finally { }
+        }
+    }",
+                GetCA3075XPathDocumentCSharpResultAt(13, 37, "TestMethod")
+            );
+
+            VerifyBasic(@"
+Imports System.Xml.XPath
+
+Class TestClass
+    Private Sub TestMethod()
+        Try
+        Catch generatedExceptionName As Exception
+            Dim xml = """"
+            Dim doc As New XPathDocument(xml)
+        Finally
+        End Try
+    End Sub
+End Class",
+                GetCA3075XPathDocumentBasicResultAt(9, 24, "TestMethod")
+            );
+        }
+
+        [Fact]
+        public void UseXPathDocumentInFinallyBlockShouldGenerateDiagnostic()
+        {
+            VerifyCSharp(@"
+using System;
+using System.Xml.XPath;
+
+class TestClass
+{
+    private void TestMethod()
+    {
+        try { }
+        catch (Exception) { throw; }
+        finally
+        {
+            var xml = """";
+            XPathDocument doc = new XPathDocument(xml);
+        }
+    }
+}",
+                GetCA3075XPathDocumentCSharpResultAt(14, 33, "TestMethod")
+            );
+
+            VerifyBasic(@"
+Imports System.Xml.XPath
+
+Class TestClass
+    Private Sub TestMethod()
+        Try
+        Catch generatedExceptionName As Exception
+            Throw
+        Finally
+            Dim xml = """"
+            Dim doc As New XPathDocument(xml)
+        End Try
+    End Sub
+End Class",
+                GetCA3075XPathDocumentBasicResultAt(11, 24, "TestMethod")
+            );
+        }
+
+        [Fact]
+        public void UseXPathDocumentInAsyncAwaitShouldGenerateDiagnostic()
+        {
+            VerifyCSharp(@"
+using System.Threading.Tasks;
+using System.Xml.XPath;
+
+class TestClass
+{
+    private async Task TestMethod()
+    {
+        await Task.Run(() => {
+            var xml = """";
+            XPathDocument doc = new XPathDocument(xml);
+        });
+    }
+
+    private async void TestMethod2()
+    {
+        await TestMethod();
+    }
+}",
+                GetCA3075XPathDocumentCSharpResultAt(11, 33, "Run")
+            );
+
+            VerifyBasic(@"
+Imports System.Threading.Tasks
+Imports System.Xml.XPath
+
+Class TestClass
+    Private Function TestMethod() As Task
+        Await Task.Run(Function() 
+        Dim xml = """"
+        Dim doc As New XPathDocument(xml)
+
+        End Function)
+    End Function
+
+    Private Sub TestMethod2()
+        Await TestMethod()
+    End Sub
+End Class",
+                GetCA3075XPathDocumentBasicResultAt(9, 20, "Run")
+            );
+        }
+
+        [Fact]
+        public void UseXPathDocumentInDelegateShouldGenerateDiagnostic()
+        {
+            VerifyCSharp(@"
+using System.Xml.XPath;
+
+class TestClass
+{
+    delegate void Del();
+
+    Del d = delegate () {
+        var xml = """";
+        XPathDocument doc = new XPathDocument(xml);
+    };
+}",
+                GetCA3075XPathDocumentCSharpResultAt(10, 29, "TestClass")
+            );
```

Comment:
Again, the warning is in an anonymous, method, and you step up the tree and find that there is no enclosing method, so you report the class name; and again, you won't need this test if you don't report the name of the "scope".